### PR TITLE
update test distributed

### PR DESCRIPTION
**Pull request by tktrungna**
Fixes #62380

* update test functions to call wheel install folder {sitepackages}/torch instead of build/ folder
* add symbolic link for shared libraries which are called by the tests (this is a bit hacky and should be fixed the rpath before compiling -- similar to https://github.com/pytorch/pytorch/blob/master/.jenkins/pytorch/test.sh#L204-L208).

### Test plan
check if all ci workflows pass

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62838
* #62797
* __->__ #62796
* #62795
* #62781

Differential Revision: [D30193142](https://our.internmc.facebook.com/intern/diff/D30193142)

### CODE REVIEW

**Comment by tktrungna** [comment]
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30193142).

**Comment by tktrungna** [comment]
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30193142).

File: test/cpp/c10d/CMakeLists.txt [hunk 1]
```diff
@@ -39,6 +49,11 @@ if(USE_CUDA)
     c10d_add_test(
       ProcessGroupNCCLErrorsTest.cpp
       torch_cpu c10d_cuda_test gtest_main __caffe2_nccl)
+    if(INSTALL_TEST)
+      install(TARGETS ProcessGroupNCCLTest DESTINATION bin)
+      install(TARGETS ProcessGroupNCCLErrorsTest DESTINATION bin)
+      install(TARGETS c10d_cuda_test DESTINATION lib)
```
Comment (zhouzhuojie) [inline test/cpp/c10d/CMakeLists.txt:55]: curious why this one is not `bin` but `lib`

**Comment by tktrungna** [comment]
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30193142).

File: test/cpp/c10d/CMakeLists.txt (hunk 1 above)
Comment (tktrungna) [reply test/cpp/c10d/CMakeLists.txt:55]: I am a cmake beginner so not sure if this is a correct solution, but basically `ProcessGroupNCCLTest` needs `libc10d_cuda_test.so` library. 
* Currently, `ProcessGroupNCCLTest` is built and put at `build/bin`, `libc10d_cuda_test.so` is compiled and put at `build/lib`
* In this PR, I've added installing step, which copies `ProcessGroupNCCLTest` to `{env}/site-packages/torch/bin`, and `libc10d_cuda_test.so` to `{env}/site-packages/torch/lib` (again would have to reluctantly add sym link to make the tests pass)

Comment (walterddr) [reply test/cpp/c10d/CMakeLists.txt:55]: i guess the name is confusing it is not exactly a test but a test util library. 
but curious where was INSTALL_TEST defined in the first place? was it in another PR?

Comment (tktrungna) [reply test/cpp/c10d/CMakeLists.txt:55]: `INSTALL_TEST` must be an option of the top-level CMakeLists.txt defined here https://github.com/pytorch/pytorch/blob/master/CMakeLists.txt#L174

**Review by zhouzhuojie — APPROVED** [review]

**Comment by tktrungna** [comment]
@tktrungna has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30193142).

**Pull request closed without merging**